### DISCUSSION TITLE
Which date format should we support?

### DISCUSSION THREAD

**Discussion by carlosthe19916** [guacsec/trustify#145] [General]
There are 2 scenarions:

### Output data
This is the data that the server generates and exposes through the API. I believe we should use ISO 8601 as it is the format used on the web.

Even if we think we should use a different format different from ISO 8601, the OUTPUT data from the server should be only ONE and only ONE, no multiple format formats.

### Input data: data send to the server through the REST API
- Import Advisories/SBOM json files: We should be able to ingest the dates of those files in the format they were created and defined by their specifications
- Filters that require dates: e.g. `GET /advisories?q="created>=DATE"`. My suggestion is to start with `ISO 8601` format as this is the format used in the web.
  - Date and time: 2011-10-05T14:48:00.000Z
  - Only date: 2018-04-13

**Comment by ctron** [comment]
I guess what you mean is RFC3339. Which is a sub-set of what ISO 8601 allows. From what I understand, most of the documents we're working with are use RFC3339 as well.

ISO 8601 allows for variants like:

* `2009-W53-7`
* `2009-W53-7T134730`
* `200704051430`
* `2024-102`

It can even have 6 digit years.

Have fun exploring: https://ijmacd.github.io/rfc3339-iso8601/

**Reply by carlosthe19916** [reply]
Thanks for point that out. Yes I meant `RFC 3339` format. I suggest using it as  it widely used in the web. But open to any other idea for sure

**Reply by bobmcwhirter** [reply]
I'd also love relative timescales when querying. `created_date > yesterday`

**Reply by carlosthe19916** [reply]
For the specific use case of dates on query parameters. I don't have anything against supporting multiple formats, as many as we want. For the UI, RFC 3339 would be lovely.

**Reply by jcrossley3** [reply]
I think #163 covers everything requested in here.
